pkg/customizations/oscap/oscap.go

```diff
@@ -98,33 +107,46 @@ func NewConfigs(oscapConfig blueprint.OpenSCAPCustomization, defaultDatastream *
 		return remediationConfig, nil, nil
 	}
 
-	tc := oscapConfig.Tailoring
-	if tc == nil {
-		return remediationConfig, nil, nil
-	}
-
-	tailoringPath := filepath.Join(DataDir, "tailoring.xml")
-	tailoredProfileID := fmt.Sprintf("%s_osbuild_tailoring", remediationConfig.ProfileID)
-
-	tailoringConfig := &TailoringConfig{
+	remediationConfig.TailoringPath = filepath.Join(DataDir, "tailoring.xml")
+	tailoringConfigs := &TailoringConfig{
 		RemediationConfig: RemediationConfig{
 			ProfileID:     remediationConfig.ProfileID,
-			TailoringPath: tailoringPath,
+			TailoringPath: remediationConfig.TailoringPath,
 			Datastream:    datastream,
 		},
-		TailoredProfileID: tailoredProfileID,
-		Selected:          tc.Selected,
-		Unselected:        tc.Unselected,
 	}
 
+	if jsonConfigs != nil {
+		if tc != nil {
+			return nil, nil, fmt.Errorf("Multiple tailoring types set, only one type can be chosen (XML/JSON/Override rules)")
+		}
+
+		if jsonConfigs.Filepath == "" {
+			return nil, nil, fmt.Errorf("Filepath to an JSON tailoring file is required")
+		}
+
+		if jsonConfigs.ProfileID == "" {
+			return nil, nil, fmt.Errorf("Tailoring profile ID is required for an JSON tailoring file")
+		}
+
+		tailoringConfigs.TailoredProfileID = jsonConfigs.ProfileID
+		tailoringConfigs.JSONFilepath = jsonConfigs.Filepath
+		remediationConfig.ProfileID = tailoringConfigs.TailoredProfileID
+
+		return remediationConfig, tailoringConfigs, nil
+	}
+
+	tailoringConfigs.Selected = tc.Selected
+	tailoringConfigs.Unselected = tc.Unselected
```

Comment:
If I'm reading the code correctly, `tc` could be `nil` at this point, no?

Reply:
I think no, but your question (and my own thoughts reading this function) make me realise this could use some refactoring.  I think a switch with separate functions for each case (xml vs json) would make things easier to read.

(var names shortened for brevity)
```go
switch {
case xml != nil && json != nil, xml != nil && tc != nil, tc != nil && json != nil:
	// return error multiple configs
case xml != nil:
	return xmlConfigs(xml, remediationConfigs)
case json != nil:
	return jsonConfigs(json, remediationConfigs)
default:
	// do the other stuff
}
```